tests/functional/test_reinstall.py

```diff
@@ -0,0 +1,204 @@
+# THIS FILE IS PART OF THE ROSE-CYLC PLUGIN FOR THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functional tests for reinstalling of config files.
+This test does the following:
+
+1. Validates a workflow.
+2. Installs a workflow with some opts set using -O and
+   ROSE_SUITE_OPT_CONF_KEYS.
+3. Re-install workflow.
+4. After modifying the source ``rose-suite.conf``, re-install the flow again.
```

Comment:
Would be be useful to have these as comments/docstrings at the beginning of each test

Reply:
Why not: I'll do that.